tutorials/intro/plot_object_annotations.py

```diff
@@ -1,245 +1,238 @@
+#    whereas :class:`~mne.Annotations` is a :class:`list`-like class defined in
+#    MNE-Python.
 #
-# :class:`~mne.Annotations` supports iterating, indexing and slicing.
-# Iterating over :class:`~mne.Annotations` and indexing with an integer returns
-# a dictionary. While slicing returns a new :class:`~mne.Annotations` instance.
 #
-# See the following examples and usages:
-
-# difference between indexing and slicing a single element
-print(annot[0])  # indexing
-print(annot[:1])  # slicing
-
-###############################################################################
-# How about iterations?
-
-for key, val in annot[0].items():  # iterate on one element which is dictionary
-    print(key, val)
+# What is a STIM channel?
+# ^^^^^^^^^^^^^^^^^^^^^^^
+#
+# :term:`STIM channels <stim channel>` (short for "stimulus channels") are
+# channels that do not receive signals from an EEG, MEG, or other sensor.
+# Instead, those channels record voltages (usually short, rectangular DC pulses
+# of fixed magnitudes sent from the experiment-controlling computer) that are
+# time-locked to experimental events, such as the onset of a stimulus or a
+# button-press response by the subject. In other cases, these pulses may not be
+# strictly time-locked to an experimental event, but instead may occur in
+# between trials to indicate the type of stimulus (or experimental condition)
+# that is about to occur on the upcoming trial.
+#
+# The DC pulses may be all on one STIM channel (in which case different
+# experimental events or trial types are encoded as different voltage
+# magnitudes), or they may be spread across several channels, in which case the
+# channel(s) on which the pulse(s) occur can be used to encode different events
+# or conditions. Even on systems with multiple STIM channels, there is often
+# one channel that records a weighted sum of the other STIM channels, in such a
+# way that voltage levels on that channel can be unambiguously decoded as
+# particular event types. On older Neuromag systems (such as that used to
+# record the sample data) this "summation channel" was typically ``STI 014``;
+# on newer systems it is more commonly ``STI101``. You can see the STIM
+# channels in the raw data file here:
+
+raw.copy().pick_types(meg=False, stim=True).plot(start=3, duration=6)
 
 ###############################################################################
-
-for idx, my_annot in enumerate(annot):  # iterate on the Annotations object
-    print('annot #{0}: onset={1}'.format(idx, my_annot['onset']))
-    print('annot #{0}: duration={1}'.format(idx, my_annot['duration']))
-    print('annot #{0}: description={1}'.format(idx, my_annot['description']))
+# You can see that ``STI 014`` (the summation channel) contains pulses of
+# different magnitudes whereas pulses on other channels have consistent
+# magnitudes. You can also see that every time there is a pulse on one of the
+# other STIM channels, there is a corresponding pulse on ``STI 014``.
```

Comment:
here's a list of all teh readers:

read_raw_artemis123
read_raw_brainvision
read_raw_bti
read_raw_cnt
read_raw_ctf
read_raw_edf
read_raw_bdf
read_raw_gdf
read_raw_eeglab
read_raw_egi
read_raw_eximia
read_raw_fieldtrip
read_raw_fif
read_raw_kit
read_raw_nicolet

Reply:
I would expect that users would know whether or not they have a STIM channel in their own data, so we wouldn't need to tell them (right?).  But if we do want to add that info, I think it should be its own page that we link to from here. (In fact I don't actually know which systems have it, since I've only ever worked with Neuromag, so I'd need help to create that page).  I'll add a TODO comment to link out to such a table whenever it gets created.